tests/benchdnn/common.hpp

```diff
@@ -107,6 +99,28 @@ extern std::string driver_name;
         } \
     } while (0)
 
+extern int verbose;
+extern bool canonical;
+extern bool mem_check;
+extern bool attr_same_pd_check;
+extern bool check_ref_impl;
+extern std::string skip_impl; /* empty or "" means skip nothing */
+extern std::string driver_name;
+
+#define LIST_OF_EXECUTION_MODES(X) \
+    X(direct) \
+    X(graph)
+
+enum class execution_mode_t {
+#define EXECUTION_MODES_ENUM(name, ...) name,
+    LIST_OF_EXECUTION_MODES(EXECUTION_MODES_ENUM)
+#undef EXECUTION_MODES_ENUM
+};
+extern execution_mode_t execution_mode;
```

Comment:
I can see why you defined the enum this way but amount of macros is freaking me out...
Could you, please, make it simple through regular `const char *` as it's done for others? I don't think this will ever get any extension...